src/js/hooks/cycleCount/useResolveStep.js

```diff
@@ -329,6 +329,62 @@ const useResolveStep = () => {
     forceRerender();
   };
 
+  const cancelCounts = async (cycleCountRequestIdsToDelete) => {
+    try {
+      show();
+      await cycleCountApi.deleteRequests(currentLocation?.id, cycleCountRequestIdsToDelete);
+      await refetchData();
+
+      // If we've canceled every product in the batch, there's no reason to stay on this screen.
+      if (tableData.current.length === 0) {
+        history.push(CYCLE_COUNT.list(TO_RESOLVE_TAB));
+      }
+    } finally {
+      hide();
+    }
+  };
+
+  const zeroRecountItemsModalButtons = (cycleCountRequestIdsToDelete) => (onClose) => ([
+    {
+      variant: 'transparent',
+      label: 'react.cycleCount.modal.zeroRecountItems.back.label',
+      defaultLabel: 'Not Now',
+      onClick: () => {
+        onClose?.();
+      },
+    },
+    {
+      variant: 'primary',
+      label: 'react.cycleCount.modal.zeroRecountItems.confirm.label',
+      defaultLabel: 'Cancel Products',
+      onClick: async () => {
+        onClose?.();
+        await cancelCounts(cycleCountRequestIdsToDelete);
+      },
+    },
+  ]);
+
+  const openZeroRecountItemsModal = (productsWithNoRecountItems) => {
+    const requestIds = productsWithNoRecountItems.map((entry) => (entry.cycleCountRequestId));
+    const productCodes = productsWithNoRecountItems.map((entry) => (entry.product));
```

Comment:
for the optimization I would suggest using `.reduce` method.
```suggestion
    const { productCodes, requestIds } = productsWithNoRecountItems.reduce((acc, entry) => ({
      requestIds: [ ...acc.requestIds, entry.cycleCountRequestId ],
      productCodes: [ ...acc.productCodes, entry.product ],
    }, { requestIds: [], productCodes: [] }))
```
I could quickly change it, but it's not "must have" change